ccl/streamingccl/streamproducer: TestStreamDeleteRange failed

ccl/streamingccl/streamproducer.TestStreamDeleteRange [failed](https://tanzanite.cluster.engflow.com/invocations/default/8c1cff8d-bd76-4a94-ba94-492fc11003b7?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3N0cmVhbWluZ2NjbC9zdHJlYW1wcm9kdWNlcjpzdHJlYW1wcm9kdWNlcl90ZXN0) on master @ [c316d6a615fa02a05357a20bf03b8a197ab27810](https://github.com/cockroachdb/cockroach/commits/c316d6a615fa02a05357a20bf03b8a197ab27810):


```
=== RUN   TestStreamDeleteRange
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStreamDeleteRange1866133490
    test_log_scope.go:81: use -show-logs to present logs inline
    replication_helpers.go:238: Replication helper seed 4190911401639640326
    replication_stream_test.go:705: consuming until 1703844974.828125620,0
    replication_stream_test.go:721: 2023-12-29 10:16:14.82848226 +0000 UTC m=+394.196193921 current frontier 0,0
    replication_stream_test.go:721: 2023-12-29 10:16:14.82865908 +0000 UTC m=+394.196370741 current frontier 1703844974.544391653,0
    replication_stream_test.go:721: 2023-12-29 10:16:14.828842809 +0000 UTC m=+394.196554471 current frontier 1703844974.544391653,0
    replication_stream_test.go:721: 2023-12-29 10:16:14.828996389 +0000 UTC m=+394.196708040 current frontier 1703844974.769984381,0
    replication_stream_test.go:721: 2023-12-29 10:16:14.863919095 +0000 UTC m=+394.231630746 current frontier 1703844974.769984381,0
    replication_stream_test.go:721: 2023-12-29 10:16:14.864154435 +0000 UTC m=+394.231866086 current frontier 1703844974.773063281,0
    replication_stream_test.go:721: 2023-12-29 10:16:15.243253389 +0000 UTC m=+394.610965050 current frontier 1703844974.773063281,0
    replication_stream_test.go:721: 2023-12-29 10:16:15.243501309 +0000 UTC m=+394.611212960 current frontier 1703844974.777176500,0
    replication_stream_test.go:721: 2023-12-29 10:16:15.243670519 +0000 UTC m=+394.611382170 current frontier 1703844974.926607949,0
    replication_stream_test.go:721: 2023-12-29 10:16:15.243796919 +0000 UTC m=+394.611508570 current frontier 1703844974.926607949,0
    replication_stream_test.go:731: consumer done
    replication_stream_test.go:705: consuming until 1703844975.340016054,0
    replication_stream_test.go:721: 2023-12-29 10:16:16.393869356 +0000 UTC m=+395.761581007 current frontier 0,0
    replication_stream_test.go:721: 2023-12-29 10:16:16.394010027 +0000 UTC m=+395.761721687 current frontier 1703844974.929626478,0
    replication_stream_test.go:721: 2023-12-29 10:16:16.394174287 +0000 UTC m=+395.761885947 current frontier 1703844974.929626478,0
    replication_stream_test.go:721: 2023-12-29 10:16:16.394299196 +0000 UTC m=+395.762010847 current frontier 1703844975.270821987,0
    replication_stream_test.go:721: 2023-12-29 10:16:16.812853792 +0000 UTC m=+396.180565454 current frontier 1703844975.270821987,0
    replication_stream_test.go:721: 2023-12-29 10:16:16.813089243 +0000 UTC m=+396.180800903 current frontier 1703844976.386307860,0
    replication_stream_test.go:731: consumer done
    replication_stream_test.go:776: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/streamingccl/streamproducer_test/pkg/ccl/streamingccl/streamproducer/replication_stream_test.go:776
        	Error:      	Not equal: 
        	            	expected: hlc.Timestamp{WallTime:1703844975340016054, Logical:0, Synthetic:false}
        	            	actual  : hlc.Timestamp{WallTime:1703844975424136095, Logical:2, Synthetic:false}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,4 @@
        	            	 (hlc.Timestamp) {
        	            	- WallTime: (int64) 1703844975340016054,
        	            	- Logical: (int32) 0,
        	            	+ WallTime: (int64) 1703844975424136095,
        	            	+ Logical: (int32) 2,
        	            	  Synthetic: (bool) false
        	Test:       	TestStreamDeleteRange
    panic.go:541: -- test log scope end --
test logs left over in: outputs.zip/logTestStreamDeleteRange1866133490
--- FAIL: TestStreamDeleteRange (33.03s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStreamDeleteRange.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-35014

--- a/pkg/ccl/streamingccl/streamproducer/replication_stream_test.go
+++ b/pkg/ccl/streamingccl/streamproducer/replication_stream_test.go
@@ -773,7 +773,7 @@ USE d;
 
 	receivedKVs, receivedDelRangeSpans := consumeUntilTimestamp(batchHLCTime)
 	require.Equal(t, t2Span.Key, receivedKVs[0].Key)
-	require.Equal(t, batchHLCTime, receivedKVs[0].Value.Timestamp)
+	require.True(t, batchHLCTime.LessEq(receivedKVs[0].Value.Timestamp))
 	require.Equal(t, expectedDelRanges, normalizeRangeKeys(receivedDelRangeSpans))
 }
 

